server/blob/blob-cloud/src/main/java/org/apache/james/blob/cloud/ContainerName.java

```diff
@@ -0,0 +1,36 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.blob.cloud;
+
+public class ContainerName {
```

Comment:
sorry for you but you know, in java you have to implement hashcode/equals/toString for that kind of class ):

Reply:
I assume lombok is not an option to avoid all the boiler plate ?

Reply:
Well, that's an entire debate. Possible alternatives are : 
* sticking with java, we could use an annotation processor like Google Autovalue (which is Effective Java recommendation) or Lombok
* we could move POJOs (and maybe some modules) to Kotlin because the transition is very smooth from our java code
* same thing with Scala

Using Lombok or Autovalue has been postponed for now because it makes IDE setup complicated and we don't want to make it harder for people to join the project, it's already hard to get started. My experience with dockerjava library is really bad for that reason.

So I'm personally convinced and very tempted to move directly to a richer language for new code and new modules. It'll probably be simpler for people to have the project to work in their IDE compared to using an annotation processor.

The downside is that it raises the bar in term of required knowledge to contribute to James and as a community project, it's always hard to choose between new stuff that will attract people and new stuff that will prevent people from contributing.

The debate is opened for some time now but we didn't conclude it. Maybe having you in the team could help us adopt the Scala solution ?